### PR TITLE
feat: subxt script to fund dev accounts via XCM

### PR DESCRIPTION
**Pull request by peterwht**
https://github.com/r0gue-io/pop-node/issues/47

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,10 +13,11 @@ members = [
     "node",
     "runtime",
     "integration-tests",
+    "scripts/fund-dev-accounts",
```
Comment (Daanvdplas) [inline Cargo.toml:16]: Perhaps exclude this one as well? It now always compiles the script when compiling the node binary which includes subxt.

File: scripts/fund-dev-accounts/main.rs [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+}
+
+use relay::*;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+	let relay_api = OnlineClient::<PolkadotConfig>::from_url("ws://127.0.0.1:8833").await?;
+
+	let dev_accounts = vec![dev::alice(), dev::bob(), dev::charlie()];
+	let fund_pop_accounts =
```
Comment (Daanvdplas) [inline scripts/fund-dev-accounts/main.rs:124]: `fund_pop_accounts_calls`

**Review by Daanvdplas — COMMENTED** [review]
Nicely done! One thing to add is three queries to Pop Network to ensure that alice, bob and charlie have received the funds.

**Comment by evilrobot-01** [comment]
Just thinking about this a little, would it not make sense to add this as a command to the `pop-node` binary? e.g. `pop-node init`.

It then becomes a first-class citizen, usable whenever a network launched, rather than compiling another side-car executable and manually launching. It could remain as its own crate, e.g. pop-node-init and be wrapped in a default feature so that it can be compiled out for releases if necessary.

I'm just thinking about the UX via `pop-cli`, where currently the node is automatically sourced and built. A node would be launched, but would not have any funds and users would not be able to deploy any contracts.

Also - have the `.scale` files been optimised to only include metadata for the pallets which are actually used? They seem large and I am sure that there is an option to trim them done to only the functionality which is used.

File: scripts/fund-dev-accounts/main.rs [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		},
+		xcm::{v3::WeightLimit, VersionedAssets, VersionedLocation},
+	};
+
+	// generate XCM message to reserve transfer funds to a designated account on
+	// Pop Parachain
+	pub(crate) fn gen_account_fund_message_call(account: Keypair) -> RuntimeCall {
+		let pop_location = VersionedLocation::V4(Location {
+			parents: 0,
+			interior: X1([Junction::Parachain(9090)]),
```
Comment (al3mart) [inline scripts/fund-dev-accounts/main.rs:35]: We might want to use the paraId set in the chainspec for pop network on rococo local here https://github.com/r0gue-io/pop-node/pull/46

`4385`

Changing the one in the zombienet file would be nice as well.

File: scripts/fund-dev-accounts/main.rs [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+			},
+			VersionedMultiAssets as VersionedAssets, VersionedMultiLocation as VersionedLocation,
+		},
+	};
+
+	// generate XCM message to reserve transfer funds to a designated account on
+	// Pop Parachain
+	pub(crate) fn gen_account_fund_message_call(account: Keypair) -> RuntimeCall {
+		let pop_location =
+			VersionedLocation::V3(Location { parents: 0, interior: X1(Junction::Parachain(9090)) });
```
Comment (al3mart) [inline scripts/fund-dev-accounts/main.rs:91]: As above, `9090` should be changed in line with what is set in the spec.

(unfortunately)

**Comment by peterwht** [comment]
> Also - have the `.scale` files been optimised to only include metadata for the pallets which are actually used? They seem large and I am sure that there is an option to trim them done to only the functionality which is used.

Unfortunately, it seems subxt has a bug with trimming metadata files (or I'm doing something wrong). Please see issue: https://github.com/paritytech/subxt/issues/1509

**Comment by peterwht** [comment]
> Just thinking about this a little, would it not make sense to add this as a command to the `pop-node` binary? e.g. `pop-node init`.

Love the idea. This will be done in a separate PR as time allows. See tracker https://github.com/r0gue-io/pop-node/issues/57

File: scripts/fund-dev-accounts/main.rs [hunk 5]
```diff
@@ -119,13 +124,18 @@ mod relay {
 }
 
 use relay::*;
+pub(crate) const AMOUNT_TO_FUND: u128 = UNIT * 1_000_000;
+
+#[subxt::subxt(runtime_metadata_path = "../metadata/pop-net.scale")]
```
Comment (evilrobot-01) [inline scripts/fund-dev-accounts/main.rs:129]: This seems an oversight?

**Review by evilrobot-01 — COMMENTED** [review]
Seems the pop interface is missing?

**Comment by Daanvdplas** [comment]
Everything works! Got an error `messageQueue.ProcessingFailed` but I think that was due `xcmPallet.VersionNotifyRequested`. The accounts were funded and the local network is set up as it should. One nitpick is the output:

Screenshot 2024-04-01 at 12 45 27

It would be nice to make this look plain and simple so that the developer can simply read what happened and doesn't get confused. E.g.:
```
XCM Reserved Transfer sent from relay.
Checking Pop Balances:
Alice: amount
Bob: amount
Charlie: amount
```
Another thing is documentation. We should explain simple and short what is happening when someone spins up a local network with Pop Network. 

Both docs & output can be done in issue #57 as well

**Review by evilrobot-01 — APPROVED** [review]
LGTM. Looking forward to seeing it as a `pop-node` command!

**Comment by evilrobot-01** [comment]
> Everything works! Got an error `messageQueue.ProcessingFailed` but I think that was due `xcmPallet.VersionNotifyRequested`. The accounts were funded and the local network is set up as it should. One nitpick is the output:

Interesting that the output includes Rococo, which should not be the case?

**Review by Daanvdplas — APPROVED** [review]

**Pull request merged**